provider/core/src/serde/mod.rs

```diff
@@ -83,6 +86,30 @@ where
 }
 
 impl DataPayload<BufferMarker> {
+    /// Deserialize a [`DataPayload`]`<`[`BufferMarker`]`>` into a [`DataPayload`] of a
+    /// specific concrete type.
+    ///
+    /// The buffer format must be known externally to this function; it is normally carried
```

Comment:
Nit: awkward phrasing. I also don't see the value of this statement. Yes the buffer format is an input, but we can already see that. That it's in the DataResponseMetadata is an implementation detail of as_deserializing.